### PR TITLE
[3.3.5] Scripts/ScarletMonastery: Mograine and Whitemane boss fight improvements

### LINKED ISSUE #15793: Scarlet Monastery/Cathedral: High Inquisitor Whitemane's door is open
TrinityCore rev. 60da7c655273 2015-10-31 16:04:14 +0100 (HEAD branch)

The door must be closed

kloster

### PR DESCRIPTION
**Pull request by Wyreth**
**Changes proposed:**

- Fix call for help when boss is aggroed (this fails when called in EnterCombat, so moved it to UpdateAI).
- Correctly handle the door to High Inquisitor Whitemane so that it opens only when supposed to.
- Fix wrong creature text for High Inquisitor Whitemane (was rehauled after WotLK, so the broadcast_text entry is not valid for WotLK).
- Whitemane should attack the player that killed Mograine first.

Closes #15793.

**Target branch(es):**

- [x] 3.3.5

**Tests performed:** works fine.

### CODE REVIEW

**Comment by Ovahlord** [comment]
Why did you set the broadcast text id = 0?

**Comment by Wyreth** [comment]
The broadcast text was changed when the instance was rehauled after WotLK. Instead of making another broadcast text, Blizzard reused that one. So it's not valid for WotLK.

**Comment by offl** [comment]
Closes #15793?

**Comment by Wyreth** [comment]
Right, thanks!

File: src/server/scripts/EasternKingdoms/ScarletMonastery/instance_scarlet_monastery.cpp [hunk 1]
```diff
@@ -117,6 +113,10 @@ class instance_scarlet_monastery : public InstanceMapScript
                             HandleGameObject(PumpkinShrineGUID, false);
                         }
                         break;
+                    case DATA_MOGRAINE_AND_WHITE_EVENT:
+                        if (state == DONE || state == IN_PROGRESS)
+                            HandleGameObject(HighInquisitorsDoorGUID, true);
```
Comment (joschiwald) [inline src/server/scripts/EasternKingdoms/ScarletMonastery/instance_scarlet_monastery.cpp]: why you are not using generic doors handling?

Comment (Wyreth) [reply src/server/scripts/EasternKingdoms/ScarletMonastery/instance_scarlet_monastery.cpp]: Generic door handling is what made it stay open all the time. I'm not sure how it should be set up for a door that should only open mid-fight.

**Comment by tkrokli** [comment]
Preferably preface SQL files with a comment line ( `--  ` ) , empty or with text.

File: src/server/scripts/EasternKingdoms/ScarletMonastery/boss_mograine_and_whitemane.cpp [hunk 2]
```diff
@@ -295,6 +303,26 @@ class boss_high_inquisitor_whitemane : public CreatureScript
                 damage = me->GetHealth() - 1;
         }
 
+        void SetGUID(ObjectGuid attacker, int32 action) override
```
Comment (tkrokli) [inline src/server/scripts/EasternKingdoms/ScarletMonastery/boss_mograine_and_whitemane.cpp]:
```
src/server/scripts/EasternKingdoms/ScarletMonastery/boss_mograine_and_whitemane.cpp:306:49:
 fatal error: unused parameter 'action' [-Wunused-parameter]
        void SetGUID(ObjectGuid attacker, int32 action) override
                                                ^
```
Comment out the unused parameter ( `int32 /*action*/` ) -- unless you plan to modify the code to use it.

Comment (Wyreth) [reply src/server/scripts/EasternKingdoms/ScarletMonastery/boss_mograine_and_whitemane.cpp]: Didn't actually mean to push this, it got pushed automatically when I pushed another branch :D gonna take this out.

File: src/server/scripts/EasternKingdoms/ScarletMonastery/instance_scarlet_monastery.cpp (hunk 1 above)
Comment (Shauren) [reply src/server/scripts/EasternKingdoms/ScarletMonastery/instance_scarlet_monastery.cpp]: By using DOOR_TYPE_SPAWN_HOLE instead of DOOR_TYPE_ROOM

Comment (Shauren) [reply src/server/scripts/EasternKingdoms/ScarletMonastery/instance_scarlet_monastery.cpp]: If it should open mid-fight and stay open after encounter then it needs manual handling

**Comment by Wyreth** [comment]
Going to scrap this for now - planning a rewrite for the whole place since the Corrupted Ashbringer event is also missing. Eventually™

File: src/server/scripts/EasternKingdoms/ScarletMonastery/instance_scarlet_monastery.cpp (hunk 1 above)
Comment (Tauriella) [reply src/server/scripts/EasternKingdoms/ScarletMonastery/instance_scarlet_monastery.cpp]: @Wyreth 
For the Door handling eventual take a look in the cMangos code.
In Cmangos, the doors work correctly.

**Comment by illfated** [comment]
Hoping to hear back from you again,, @Wyreth (especially regarding any plans to work on this). 😃

**Pull request closed without merging**